### PR TITLE
[WIP]---MAKES PEACEKEEPER CYBORGS NOT WORTHLESS---

### PR DESCRIPTION
**Pull request by silicons**
###  I WILL BE CODING IN EMP AND FLASH PROTECTION BUILT INTO THE SHIELDS, AS VARIABLES. THE PEACEKEEPER CYBORG VARIENT OF THE SHIELD WILL NOT HAVE EMP PROTECTION, AS THIS IS MOSTLY FOR ADMIN GIMMICKS/VAREDITING AND FUTURE EXPANSIONS (IE syndiborgs). HOWEVER, I WANT TO CONSIDER GIVING IT FLASH PROTECTION, SO IF YOU FLASH IT WITH SHIELDS ON IT WILL BE BLINDED, BUT NOT STUNNED. RIGHT NOW EACH FLASH WILL TAKE MORE OR LESS 500 POWER TO BLOCK AND THE PROTECTION WILL SHORT OUT AFTER 7 FLASHES (considering as a melee weapon they're really quick to apply). OPINIONS?

+PK borgs now get an energy bola launcher, however this time, it has a 3 second delay, costs 150 energy, and autoremoves 2.5 seconds after it applies to someone. It has slow projectile speed.
+PK borgs now get an integrated repair droid by default. However, this means it can't get a normal repair droid, and the integrated one is faster (2x), but consumes a large amount of energy (5400 per 200 health, could be higher if that's not weak enough). Won't work while emagged for balance. 
+WIP: PK borgs get a shield module that takes 2 seconds of not moving to activate, and 2 seconds of not moving to deactivate, and prevents any other module from being used while activated, but reduces damage from projectiles by 70% and melee by 50%. Impacts drain power at a high rate, maintaining shield drains power by a high rate. Blocks flashes. By high I mean from anywhere from 100 power per hit to "10 lasers will drain 5000-7000 power". Stats/functions are not final, maybe a module that protects someone adjacent instead?
=None of these have emagged functions
Discuss and yell at me.
None of these are final, for one the ebola is probably getting even harder nerfs by 3 second delay and autoremove 1.5-2 seconds instead.

:cl: Mekhi
rscadd: Peacekeeper Cyborgs get an energy bola launcher again. Long recharge cycle, high power cost, and energy bolas fired will fall off after 2.5 seconds.
rscadd: Peacekeeper Cyborgs get an inefficient but fast integrated repair droid. Won't work while emagged for balance.
rscadd: Peacekeeper Cyborgs have an energy shield module to actively reduce damage with a very high power drain, and disallows other modules from being used.
\:cl:

https://tgstation13.org/phpBB/viewtopic.php?f=9&t=8542
TL;DR: MAKES PK BORGS INTO MACHINES THAT CAN QUICKLY DISPELL A RIOT OR BREAK UP A FIGHT, BUT USING HUGE AMOUNTS OF ENERGY WHILE DOING SO. NOWHERE NEAR FINAL, WONT BE WORKED ON AS ACTIVELY AS I TRY TO RUSH OFF FLIGHTPACKS.

### CODE REVIEW

**Comment by xxalpha** [comment]
You could stop trying to reinvent the wheel and just re add the secborg.

**Comment by kevinz000** [comment]
@xxalpha They were never removed they're disabled in the server config.
Infact open a test server on master branch and it'll have it enabled.
These are never meant to be secborgs, I'm trying to make them not effective in chasing people down.

**Comment by xxalpha** [comment]
>These are never meant to be secborgs

People can keep trying to tell themselves that but I know what's up.

Just re add the secborgs if you're going to buff the pk.

**Comment by kevinz000** [comment]
>no hard stuns
>most abilities are more for cockblocking
Pls.

**Comment by jughu** [comment]
idk why secborg was ever removed it was fine just give it just a baton only and zipties flash and no taser shit make it melee tier

**Comment by kevinz000** [comment]
The difference is you can send a secborg and it can stunlock a crowd unless someone has a flash and gets a lucky hit.
You can send a peacekeeper borg into a crowd and it'll get beat up but everyone will be confused, dizzy, and slowdown'd. Maybe even a few barriers and cookies tossed around :^)

**Comment by xxalpha** [comment]
>The difference is you can send a secborg and it can stunlock a crowd unless someone has a flash and gets a lucky hit.

A lucky flash hit? Just AOE. Cyborgs are ridiculously easy to stop.

**Comment by collinxchu** [comment]
I like these changes, personally. I wish there were better debuffs in SS13 to apply beyond the ones that we DO have, but eh. This works well enough.

**Comment by Supermichael777** [comment]
make the hacked energy bola launcher be attackable to max the launch force. YOU CANT RUN WITH NO LEGS

**Comment by Cheridan** [comment]
the thing is, PK borgs are _supposed_ to be useless.

**Comment by RemieRichards** [comment]
@Cheridan *at valids*
They're supposed to be good at ending combat peacefully (usually by being a punching bag)

**Comment by Anonmare** [comment]
They're literally meant to be crowd control bots, they're more like walking, talking punching bags who annoy people into giving up fighting

**Comment by TechnoAlchemisto** [comment]
The intention to get people to stop fighting makes sense but more often then not their attempts at intervention (back when they weren't nerfed) ended up getting people killed.

Being annoying isn't a good mechanic

**Comment by Dorsisdwarf** [comment]
We need the Peacekeeper II. Its only module is the secbarrier, it comes with VTECH, and it has 500 health.

**Comment by Anonmare** [comment]
@TechnoAlchemisto Explain the Clown then lad.

**Comment by jughu** [comment]
Bolas was shit anyways because if i loke use dizzy chems on a guy that puts the other guy in a fighting advemtage and thry womt stop fighting untill one side does or sec interferes there is no peace

**Comment by TechnoAlchemisto** [comment]
@Anonmare clowns aren't obligated to slip you when you're fighting a traitor with lethals

**Comment by AdamElTablawy** [comment]
@TechnoAlchemisto

Any clown who doesn't slip you at crucial life threatening moments is a shit clown

I had a clown pie me in the face as captain as I was fighting with a WAR!op. That was a good clown.

also kevinz the thing is that peacekeeper borgs really can't bust up a fight between two crew let alone an angry traitor without getting bored and dragging beepsky or an unsecured chem dispenser and fastmixing lube smoke

replace their crappy dizzy chem with cryptobiolin and replace their tiring solution with a chem that adds heavy slowdown instead, give them zipties to cuff people who end up falling down, I dunno

tbh at this point the only way i can stop harm as a peacekeeper is firing myself out of elaborate disposal cannons into harm causers, knocking them over and then placing fifty cookies on top of them

File: code/modules/projectiles/projectile/energy.dm [hunk 1]
```diff
@@ -109,28 +109,27 @@
 	new/obj/item/weapon/restraints/legcuffs/beartrap/energy(loc)
 	..()
 
-/obj/item/projectile/energy/trap/cyborg
+/obj/item/projectile/energy/cyborgtrap
 	name = "Energy Bola"
-	icon_state = "e_snare"
+	icon_state = "ebola"
```
Comment (AdamElTablawy) [inline code/modules/projectiles/projectile/energy.dm]: ebola

Comment (kevinz000) [inline code/modules/projectiles/projectile/energy.dm]: working as intended sir

**Comment by kevinz000** [comment]
@AdamElTablawy @Cheridan If players start being shits with beepsky I could just disable PKborg/borgs as a whole from dragging beepsky/EDs

**Comment by kevinz000** [comment]
@xxalpha FLASHES DO NOT AOE STUN BORGS. I hope you didn't mean that because basically everyone knows this right?

**Comment by kevinz000** [comment]
also if a traitor wants a pkborg to stop being annoying, flash and emag, revolver it to death as 4 revolver shots will kill and 3 will take all modules, throw their infinite emp grenades at it, or just blow it.

**Comment by optimumtact** [comment]
no stuns by default on borgs ever again.

**Comment by Boredone** [comment]
I don't see any stuns though? Closest is the Bola, which I believe would just slow folks down?

**Comment by optimumtact** [comment]
turns out it's not a stun

**Comment by kevinz000** [comment]
i know this does not compile btw don't yell at me it's just a missing proc that i'm yet to code.

**Comment by kevinz000** [comment]
if you didn't already I updated the description of the PR, please read the top.

**Comment by kevinz000** [comment]
Back to active development.

**Comment by TehZombehz** [comment]
[Yelling]
